File: CS/PopupSizeExample.Module.Win/Controllers/CustomizeFormSizeController.cs
using System;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Templates;
using PopupSizeExample.Module.BusinessObjects;

namespace PopupSizeExample.Module.Win.Controllers {
    public class CustomizeFormSizeController : WindowController {
        protected override void OnActivated() {
            base.OnActivated();
            //Subscribe to this or the XafApplication.CustomizeTemplate event from a Controller or other places to access the target window template.
            Window.TemplateChanged += Window_TemplateChanged;
        }
        private void Window_TemplateChanged(object sender, EventArgs e) {
            if(Window.Template is System.Windows.Forms.Form && Window.Template is ISupportStoreSettings) {
                //Handle this or the Form.HandleCreated/Load events for applying customizations AFTER the default XAF template settings were applied.
                ((ISupportStoreSettings)Window.Template).SettingsReloaded += OnFormReadyForCustomizations;
            }
        }
        private void OnFormReadyForCustomizations(object sender, EventArgs e) {
            if(YourCustomBusinessCondition(Window.View)) {
                //Here we are resizing the form based on the View object properties. You can also adjust other Form settings as well.
                ((System.Windows.Forms.Form)sender).Size = ((IFormSizeProvider)Window.View.CurrentObject).GetFormSize();
            }
        }
        private bool YourCustomBusinessCondition(View view) {
            return view != null && view.CurrentObject is IFormSizeProvider;
        }
        protected override void OnDeactivated() {
            Window.TemplateChanged -= Window_TemplateChanged;
            base.OnDeactivated();
        }
    }
}

File: VB/PopupSizeExample.Module.Win/Controllers/CustomizeFormSizeController.vb
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Templates
Imports PopupSizeExample.Module.BusinessObjects

Namespace PopupSizeExample.Module.Win.Controllers
    Public Class CustomizeFormSizeController
        Inherits WindowController

        Protected Overrides Sub OnActivated()
            MyBase.OnActivated()
            'Subscribe to this or the XafApplication.CustomizeTemplate event from a Controller or other places to access the target window template.
            AddHandler Window.TemplateChanged, AddressOf Window_TemplateChanged
        End Sub
        Private Sub Window_TemplateChanged(ByVal sender As Object, ByVal e As EventArgs)
            If TypeOf Window.Template Is System.Windows.Forms.Form AndAlso TypeOf Window.Template Is ISupportStoreSettings Then
                'Handle this or the Form.HandleCreated/Load events for applying customizations AFTER the default XAF template settings were applied.
                AddHandler DirectCast(Window.Template, ISupportStoreSettings).SettingsReloaded, AddressOf OnFormReadyForCustomizations
            End If
        End Sub
        Private Sub OnFormReadyForCustomizations(ByVal sender As Object, ByVal e As EventArgs)
            If YourCustomBusinessCondition(Window.View) Then
                'Here we are resizing the form based on the View object properties. You can also adjust other Form settings as well.
                DirectCast(sender, System.Windows.Forms.Form).Size = DirectCast(Window.View.CurrentObject, IFormSizeProvider).GetFormSize()
            End If
        End Sub
        Private Function YourCustomBusinessCondition(ByVal view As View) As Boolean
            Return view IsNot Nothing AndAlso TypeOf view.CurrentObject Is IFormSizeProvider
        End Function
        Protected Overrides Sub OnDeactivated()
            RemoveHandler Window.TemplateChanged, AddressOf Window_TemplateChanged
            MyBase.OnDeactivated()
        End Sub
    End Class
End Namespace
